### ISSUE TITLE
Box shadow: Adjust colors to system theme? 

### ISSUE THREAD

**Issue by RedBearAK** [neuromorph/custom-osd#2]
Once upon a time, I experimented with changing drop shadows to white/gray or even a bright color, for helping visually separate windows on a dark theme. I still don't understand why this isn't a thing with dark themes in general. The darker the theme, the more the edges of windows blend together on the screen. It can make it quite difficult at times to even find a new small window or dialog that pops up on a screen full of other windows. 

Here's what I managed for a brief time, before the effect suddenly stopped working, and I could never get it working again. 

https://www.reddit.com/r/gnome/comments/nwcnt1/turn_window_drop_shadows_into_under_lighting_for/

If it were possible to set the box shadow to be a custom color, or have it use different custom colors based on the system theme state, that would be pretty neat. I don't do this, but there are a lot of users who like to change their theme automatically from light to dark when it gets dark outside. If Custom OSD could adjust in that way also, changing the background color/border and even the drop shadow when the theme changes, I think a lot of users would really like that. 

I would like to see at least the ability to manually change the drop shadow color/opacity. But if this is too involved feel free to say so and close this.

**Comment by neuromorph** [comment]
Great point in general for dark themes. Windows blending together without proper boundaries in dark themes is one of my pet-peeves as well (even browser tabs tend to merge). As for OSD, the separation or blending is personal choice. More transparency will cause more merging with background. To separate, best option is to turn on Border. I tried with shadow but it didn't look good. Automatically matching with the theme will be more involved. Maybe, an option to auto detect theme color, and apply it to the widgets, can be considered for future update.  
As for manually changing the shadow, you can do that. You can also apply the same kind of shadow style as you shared in the link. Instructions are in ReadME, copying below. Just uncomment box-shadow line in that file and use custom size numbers and color:  

For optional styling of advanced css options that are not in settings, you will need to edit "spreadsheet.css" file at -  
~/.local/share/gnome-shell/extensions/custom-osd@neuromorph/  
 This allows for some esoteric tinkering for the ones so inclined. 

Example code:
```
.osd-style {
    padding-top: 5px;
    padding-bottom: 5px;
    box-shadow: 1px 1px 5px grey;
}
```

**Comment by RedBearAK** [comment]
@neuromorph 

> I tried with shadow but it didn't look good. Automatically matching with the theme will be more involved. Maybe, an option to auto detect theme color, and apply it to the widgets, can be considered for future update.

Yes, I tried quite a few different things when I was experimenting, and a lot of them really didn't look amazing. They could be a bit overwhelming. But a nice white or light gray was OK, usually. 

My idea was more about just letting the user pick a shadow/border/background color for "light" themes and "dark" themes and then just switching it based on whether the system says it is in "light" mode or "dark" mode. A number of GTK apps recently have been having a "Follow system theme" option. I'm not sure that does anything more than checking for light vs dark. But I could be wrong. 

Thanks for the tip of what to look at to play with the effect.
